spec/features/memberships/sac_terminate_membership_wizard_spec.rb

```diff
@@ -0,0 +1,136 @@
+# frozen_string_literal: true
+
+#  Copyright (c) 2024, Schweizer Alpen-Club. This file is part of
+#  hitobito_sac_cas and licensed under the Affero General Public License version 3
+#  or later. See the COPYING file at the top-level directory or at
+#  https://github.com/hitobito/hitobito_sac_cas.
+
+require "spec_helper"
+
+describe "terminate sac membership wizard", js: true do
+  let(:person) { people(:mitglied) }
+  let(:role) { person.roles.first }
+  let(:operator) { person }
+  let(:group) { groups(:bluemlisalp_mitglieder) }
+  let(:termination_reason) { termination_reasons(:deceased) }
+
+  before do
+    sign_in(operator)
+  end
+
+  context "as SAC Mitarbeiter" do
+    let(:operator) { people(:admin) }
+
+    xit "can execute wizard with immediate deletion" do
+      visit history_group_person_path(group_id: group.id, id: person.id)
+      within("#role_#{role.id}") do
+        click_link "Austritt"
+      end
+      choose "Sofort"
+      click_button "Weiter"
+      select termination_reason.text
+      check "Newsletter beibehalten"
+      check "Meine Daten sollen nach dem Austritt erhalten blieben und ich kann für Spendenaufrufe kontaktiert werden"
+      check "Ich möchte weiterhin über Spendenaktionen informiert werden."
+      expect do
+        click_button "Austritt beantragen"
+        expect(page).to have_content "Deine SAC-Mitgliedschaft wurde gekündet."
+        role.reload
+      end
+        .to change { person.roles.count }.by(-1)
+        .and change { role.deleted_at }.from(nil)
+        .and change { role.termination_reason }.from(nil).to(termination_reason)
+    end
+
+    it "can execute wizard at end of year" do
+      visit history_group_person_path(group_id: group.id, id: person.id)
+      within("#role_#{role.id}") do
+        click_link "Austritt"
+      end
+      expect(page).to have_title "SAC-Mitgliedschaft beenden"
+      choose "Auf 31.12.#{Date.current.year}"
+      click_button "Weiter"
+      select termination_reason.text
+      check "Newsletter beibehalten"
+      check "Meine Daten sollen nach dem Austritt erhalten blieben und ich kann für Spendenaufrufe kontaktiert werden"
+      check "Ich möchte weiterhin über Spendenaktionen informiert werden."
+      expect do
+        click_button "Austritt beantragen"
+        expect(page).to have_content "Deine SAC-Mitgliedschaft wurde gekündet."
+        role.reload
+      end
+        .to change { role.termination_reason }.from(nil).to(termination_reason)
+      # TODO: role delete_on should change to end of year?
+      # .and change { role.delete_on }.to(Date.new(Date.current.year, 12, 31))
```

Comment:
```suggestion
        .to change { role.termination_reason }.from(nil).to(termination_reason)
        .and change { role.delete_on }.to(Date.new(Date.current.year, 12, 31))
```

With  https://github.com/hitobito/hitobito_sac_cas/pull/748  merged this can be removed.